stablehlo/transforms/VhloToVersion.cpp

```diff
@@ -181,7 +181,14 @@ bool isLegalOperation(Operation* op, const Version& targetVersion) {
   auto opInterface = dyn_cast<VersionedOpInterface>(op);
   if (!opInterface) return false;
   if (!isLegalVersion(opInterface, targetVersion)) return false;
-  LLVM_DEBUG(llvm::dbgs() << "Legal version for target. " << op << '\n');
+  LLVM_DEBUG(llvm::dbgs() << "Legal op version for target. " << op << '\n');
+
+  // Validate op constraints
+  auto constraintInterface = dyn_cast<VersionedOpConstraintInterface>(op);
+  if (constraintInterface &&
+      failed(constraintInterface.validateConstraint(op, targetVersion)))
+    return false;
+  LLVM_DEBUG(llvm::dbgs() << "Legal constraints for target. " << op << '\n');
```

Comment:
Correction to my own code here - I'm realizing that this will be difficult to debug what's going wrong as is. Adding a debug message the denote failed constraint verification will make this simpler.

```
  if (!isLegalVersion(opInterface, targetVersion)) return false;
  auto constraintInterface = dyn_cast<VersionedOpConstraintInterface>(op);
  if (constraintInterface &&
      failed(constraintInterface.validateConstraint(op, targetVersion))) {
    LLVM_DEBUG(llvm::dbgs() << "Op failed to satisfy versioned constraints. " << op << '\n');
    return false;
  }
  LLVM_DEBUG(llvm::dbgs() << "Legal op version for target. " << op << '\n');

```